test/e2e/operator_test.go

```diff
@@ -0,0 +1,290 @@
+		os.Exit(1)
+	}
+	defaultParentZoneID = os.Getenv("EXTDNS_PARENT_ZONEID")
+	if defaultParentDomainName == "" {
+		fmt.Printf("No parent zone ID specified. Please set the environment variable EXTDNS_PARENT_ZONEID\n")
+		os.Exit(1)
+	}
+
+	domainName = generateDomainName()
+
+	platform := infraConfig.Status.PlatformStatus
+	if platform == nil {
+		fmt.Printf("platform status is missing for infrastructure %s", infraConfig.Name)
+		os.Exit(1)
+	}
+	switch platform.Type {
+	case configv1.AWSPlatformType:
+		mySession := awsSession.Must(awsSession.NewSession())
+		dnsClient.aws = route53.New(mySession)
+	default:
+		fmt.Printf("Unsupported Provider")
+		os.Exit(1)
+	}
+
+	dialer = &net.Dialer{
+		Resolver: &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: 5000 * time.Millisecond,
+				}
+				return d.DialContext(ctx, "udp", "8.8.8.8:53")
+			},
+		},
+	}
+
+	httpClient = &http.Client{
+		Transport: &http.Transport{
+			Dial:        dialer.Dial,
+			DialContext: dialer.DialContext,
+		},
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestOperatorAvailable(t *testing.T) {
+	expected := []appsv1.DeploymentCondition{
+		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+	}
+	if err := waitForOperatorDeploymentStatusCondition(t, kclient, expected...); err != nil {
+		t.Errorf("did not get expected available condition: %v", err)
+	}
+}
+
+func TestExternalDNSRecordLifecycle(t *testing.T) {
+	if err := kclient.Create(context.TODO(), &defaultNamespace); err != nil {
+		t.Fatalf("Failed to create default namespace: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), &defaultNamespace)
+	credsSecret, err := createCredsSecret(t, kclient)
+	if err != nil {
+		t.Fatalf("Failed to create credentials secret: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), credsSecret)
+	hostedZoneID, err := createHostedZone(t, domainName, defaultParentDomainName, defaultParentZoneID)
+	if err != nil {
+		t.Fatalf("Failed to create hosted zone for tests: %v", err)
+	}
+	defer deleteHostedZone(t, hostedZoneID, domainName, defaultParentZoneID)
+	extDNS := defaultExternalDNS(t, hostedZoneID, credsSecret)
+	if err := kclient.Create(context.TODO(), &extDNS); err != nil {
+		t.Fatalf("Failed to create external DNS: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), &extDNS)
+	if err := kclient.Create(context.TODO(), &defaultBackend); err != nil {
+		t.Fatalf("Failed to create default backend: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), &defaultBackend)
+	if err := kclient.Create(context.TODO(), &defaultService); err != nil {
+		t.Fatalf("Failed to create default service: %v", err)
+	}
+	defer kclient.Delete(context.TODO(), &defaultService)
+	expectedDomainName := fmt.Sprintf("%s.%s", defaultService.ObjectMeta.Name, domainName)
+	err = wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
would be nice to have this timeout as a variable too, somewhere at the beginning of the file for readability/ease of configuration.